onmt/trainer.py

```diff
@@ -70,7 +70,8 @@ def build_trainer(opt, device_id, model, fields, optim, model_saver=None):
                            model_dtype=opt.model_dtype,
                            earlystopper=earlystopper,
                            dropout=dropout,
-                           dropout_steps=dropout_steps)
+                           dropout_steps=dropout_steps,
+                           encode_tgt=True if opt.lambda_cosine > 0 else False)
```

Comment:
are you sure it wont raise an issue for current checkpoint where `opt.lambda_cosine` does not exist?!

Reply:
Should be handled here:
https://github.com/OpenNMT/OpenNMT-py/blob/5ea9cb2fa0d12e1f8b3b835ecf51a4b949fc0a14/onmt/utils/parse.py#L84-L90